app/models/manageiq/providers/redhat/builder.rb

```diff
@@ -0,0 +1,48 @@
+class ManageIQ::Providers::Redhat::Builder
+  class << self
+    def build_inventory(ems, target)
+      case target
+      when ManageIQ::Providers::Redhat::InfraManager
```

Comment:
It could be done as you suggest. This piece is based on the logic from other providers.
case graph_full:
case graph_target:
else regualr_refresh

To keep this clarity we would need to have if elseif and else which is not that different.